code/hash/dig-vs-[]-vs-fetch.rb

```diff
@@ -0,0 +1,31 @@
+require 'benchmark/ips'
+
+h = { a: { b: { c: { d: { e: "foo" } } } } }
+
+Benchmark.ips do |x|
+  x.report 'Hash#dig' do
+    h.dig(:a, :b, :c, :d, :e)
+  end
+
+  x.report 'Hash#[]' do
+    h[:a][:b][:c][:d][:e]
+  end
+
+  x.report 'Hash#[] ||' do
+    ((((h[:a] || {})[:b] || {})[:c] || {})[:d] || {})[:e]
+  end
+
+  x.report 'Hash#[] &&' do
+    h[:a] && h[:a][:b] && h[:a][:b][:c] && h[:a][:b][:c][:d] && h[:a][:b][:c][:d][:e]
+  end
+
+  x.report 'Hash#fetch' do
+    h.fetch(:a).fetch(:b).fetch(:c).fetch(:d).fetch(:e)
+  end
+
+  x.report 'Hash#fetch fallback' do
+    h.fetch(:a, {}).fetch(:b, {}).fetch(:c, {}).fetch(:d, {}).fetch(:e, nil)
```

Comment:
@mblumtritt in fact it will not. Why would it create Proc?

In any case, @dideler I tend to agree that probably some people use `fetch(key, {})`.
And that looks pretty fine, but in this case it worth to "cache" that object:

``` ruby
o = {}
h.fetch(:a, o).fetch(:b, o)
```

That's just my thoughts. I don't insist on anything :D